src/sentry/adoption/manager.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+
+from collections import namedtuple
+
+FEATURE_LOCATION = set(['language', 'integration', 'code', 'web', 'admin'])
+
+Feature = namedtuple('Feature', ['id', 'slug', 'name', 'location', 'prerequesite'])
+
+
+class AdoptionManager(object):
+    def __init__(self):
+        self._id_registry = {}
+        self._slug_registry = {}
+        self._integration_slugs = {}
+        self._location_slugs = {}
```

Comment:
If you make this a `defaultdict(set)` you don't have to do the `setdefault` stuff below in `__init__`.